make/hotspot/lib/JvmFeatures.gmk

```diff
@@ -169,8 +169,12 @@ ifeq ($(call check-jvm-feature, link-time-opt), true)
   # later on if desired
   JVM_OPTIMIZATION := HIGHEST_JVM
   ifeq ($(call isCompiler, gcc), true)
-    JVM_CFLAGS_FEATURES += -flto -fuse-linker-plugin
-    JVM_LDFLAGS_FEATURES += -flto -fuse-linker-plugin -fno-strict-aliasing
+    JVM_CFLAGS_FEATURES += -flto=$(JOBS) -fuse-linker-plugin -fno-strict-aliasing -fno-fat-lto-objects
+    JVM_LDFLAGS_FEATURES += -O3 -flto=$(JOBS) -fuse-linker-plugin -fno-strict-aliasing
```

Comment:
Shouldn't the `-O3` be using the `OPTIMIZATION` level in case it has been overridden from `HIGHEST_JVM`?

Reply:
It may make sense to have the `-O*` argument to the linker match what we do for compilation. I'm not sure how related they are.

Reply:
Good catch, thanks. Will also add this flag to the opt-size option so it can override what link-time-opt has set

Reply:
I stand corrected, it was not actually needed for opt-size